docs/01-app/01-getting-started/07-caching-and-revalidating.mdx

```diff
@@ -0,0 +1,299 @@
+import { getPosts } from '@/app/lib/data'
+
+export default async function Page() {
+  'use cache'
+  const posts = await getPosts()
+
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+```js filename="app/blog/page.js" highlight={4} switcher
+import { getPosts } from '@/app/lib/data'
+
+export default async function Page() {
+  'use cache'
+  const posts = await getPosts()
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+Alternatively, you can add the `'use cache'` directive at the top of the file to mark all components in the file as cacheable:
+
+```tsx filename="app/blog/page.tsx" highlight={1} switcher
+'use cache'
+
+import { getPosts } from '@/app/lib/data'
+
+export default async function Page() {
+  const posts = await getPosts()
+
+  return <Posts posts={posts} />
+}
+
+async function Posts({ posts }) {
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+```js filename="app/blog/page.js" highlight={1} switcher
+'use cache'
+
+import { getPosts } from '@/app/lib/data'
+
+export default async function Page() {
+  const posts = await getPosts()
+
+  return <Posts posts={posts} />
+}
+
+async function Posts({ posts }) {
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+### Functions
+
+You can cache the return value of an **asynchronous** function, including data requests, by adding the `'use cache'` directive inside the function body:
+
+```ts filename="app/lib/data.ts" switcher
+export async function getPosts(slug: string) {
+  'use cache'
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+```js filename="app/lib/data.js" switcher
+export async function getPosts(slug) {
+  'use cache'
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+With the approach above, you can call the function throughout your application code and the same cache entry will be reused as long as the arguments are the same.
+
+Alternatively, you can also cache all functions within a file by adding the `'use cache'` directive at the top of the file:
+
+```ts filename="app/lib/data.ts" switcher
+'use cache'
+
+export async function getPosts() {
+  const data = await fetch(`/api/posts`)
+  return data.json()
+}
+
+export async function getPostBySlug(slug: string) {
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
+```js filename="app/lib/data.js" switcher
+'use cache'
+
+export async function getPosts() {
+  const data = await fetch(`/api/posts/`)
+  return data.json()
+}
+
+export async function getPostBySlug(slug: string) {
+  const data = await fetch(`/api/posts/${slug}`)
+  return data.json()
+}
+```
+
```

Comment:
Maybe again: There will be two separate entries into the cache. One for `getPosts` and another for `getPostsBySlug`, where the cache will be invalidated if the `slug` argument changes.

Reply:
For the invalidated part, wouldn't it be two separate entries? A cache key is the function's unique identity + the input.